scripts/change-log-generator.js

```diff
@@ -236,29 +257,79 @@ function filterExistingPREntries(parsedCommits) {
  */
 function getMessagesGroupedByPackage(parsedCommits) {
   var groupedMessages = {};
+  var sortedMessages = {};
   parsedCommits.forEach(function(map) {
     map[PACKAGES].forEach(function(packageName) {
-      groupedMessages[packageName] = groupedMessages[packageName] || [];
-      groupedMessages[packageName].push(
-        util.format(MESSAGE_FORMAT, map[MESSAGE], map[PR_NUM], map[PR_NUM])
-      );
+      var key = generateKey(packageName, map[TYPE]);
+      if (key) {
+        groupedMessages[key] = groupedMessages[key] || [];
+        groupedMessages[key].push(
+          util.format(MESSAGE_FORMAT, map[MESSAGE], map[PR_NUM], map[PR_NUM])
+        );
+      }
     });
   });
+  Object.keys(groupedMessages)
+    .sort()
+    .forEach(function(key) {
+      sortedMessages[key] = groupedMessages[key];
+    });
   if (ADD_VERBOSE_LOGGING) {
-    console.log('\nMessages grouped by package:');
-    console.log(groupedMessages);
+    console.log('\nStep 5: Group results by type and package name.');
+    console.log('Sorted messages by package name and type:');
+    console.log(sortedMessages);
   }
-  return groupedMessages;
+  return sortedMessages;
+}
+
+/**
+ * Generate the key to be used in the grouped messages map. This will help us
+ * determine whether this is an addition or fix, along with the package header
+ * that the commit should be inserted under.
+ *
+ * If we have a type that should be ignored, return an empty key.
+ */
+function generateKey(packageName, type) {
+  const typesToIgnore = [
+    'chore',
+    'style',
+    'refactor',
+    'test',
+    'build',
+    'ci',
+    'revert'
+  ];
+  if (typesToIgnore.includes(type)) {
+    return '';
+  }
+  var key = packageName;
+  if (type) {
+    if (type == 'feat') {
+      key = 'Added|' + packageName;
+    } else {
+      key = 'Fixed|' + packageName;
+    }
+  } else {
+    key = 'Fixed|' + packageName; // If no type is specified, assume this is a fix.
+  }
+  return key;
 }
 
 function getChangeLogText(releaseBranch, groupedMessages) {
   var changeLogText = util.format(
     LOG_HEADER,
     releaseBranch.toString().replace('origin/release/v', '')
   );
-  Object.keys(groupedMessages).forEach(function(packageName) {
+  var lastType = '';
+  Object.keys(groupedMessages).forEach(function(typeAndPackageName) {
+    var type = typeAndPackageName.split('|')[0];
+    var packageName = typeAndPackageName.split('|')[1];
```

Comment:
```suggestion
    var [type, packageName] = typeAndPackageName.split('|');
```
handy way to split and assign